### PR TITLE
HW12

### PR DESCRIPTION
**Pull request by DVRudoi**
HW12

### CODE REVIEW

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+                    By.xpath("//tr[" + index + "]/td[1]")).getText().contains("Java (")
+                    || getDriver().findElement(
+                    By.xpath("//tr[" + index + "]/td[1]")).getText().equals("Java")) {
+                if (numberOfComments < Integer.parseInt(getDriver().findElement(
+                        By.xpath("//tr[" + index + "]/td[4]")).getText())){
+                    numberOfComments = Integer.parseInt(getDriver().findElement(
+                            By.xpath("//tr[" + index + "]/td[4]")).getText());
+                    actualResult = getDriver().findElement(
+                            By.xpath("//tr[" + index + "]/td[1]")).getText();
+
```
Comment (anonymoustuna) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:279]: You might delete this empty line.

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                }
+            }
+        }
+
+        Assert.assertEquals(actualResult, expectedResult);
+    }
+
```
Comment (anonymoustuna) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:286]: Empty line.

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+}
+
```
Comment (anonymoustuna) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:288]: Empty line.

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        WebElement buttonBrowseLanguages = getDriver().findElement(
+                By.xpath("//li/a[@href = '/abc.html']"));
+        buttonBrowseLanguages.click();
+
+        WebElement buttonLetterM = getDriver().findElement(
+                By.xpath("//a[@href = 'm.html']"));
+        buttonLetterM.click();
+
+        String actualAuthor = getDriver()
+                .findElement(By.xpath("//tr[22]/td[2]"))
```
Comment (ViktorPodgornov) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:86]: Use it. We talked about it yesterday. //a[text()='Mathematica']//ancestor::tr/td[2]

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+                .getText();
+
+        String actualDate = getDriver()
+                .findElement(By.xpath("//tr[22]/td[3]"))
```
Comment (ViktorPodgornov) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:90]: //a[text()='Mathematica']//ancestor::tr/td[3]

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+                .getText();
+
+        String actualComments = getDriver()
+                .findElement(By.xpath("//tr[22]/td[4]"))
```
Comment (ViktorPodgornov) [inline src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:94]: //a[text()='Mathematica']//ancestor::tr/td[4]

**Review by ViktorPodgornov — APPROVED** [review]

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java (hunk 5 above)
Comment (DVRudoi) [reply src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:90]: Thanks

File: src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java (hunk 1 above)
Comment (DVRudoi) [reply src/test/java/qa_java_beginners/DVRudoi99BottlesOfBeerTest.java:279]: Thanks

**Pull request merged**